manifest/ociartifact/artifact_manifest.go

```diff
@@ -0,0 +1,144 @@
+package ociartifact
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/distribution/distribution/v3"
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func init() {
+	artifactFunc := func(b []byte) (distribution.Manifest, distribution.Descriptor, error) {
+		if err := validateArtifactManifest(b); err != nil {
+			return nil, distribution.Descriptor{}, err
+		}
+		m := new(DeserializedArtifactManifest)
+		err := m.UnmarshalJSON(b)
+		if err != nil {
+			return nil, distribution.Descriptor{}, err
+		}
+
+		dgst := digest.FromBytes(b)
+		return m, distribution.Descriptor{Digest: dgst, Size: int64(len(b)), MediaType: v1.MediaTypeArtifactManifest}, err
+	}
+	err := distribution.RegisterManifestSchema(v1.MediaTypeArtifactManifest, artifactFunc)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to register artifact manifest: %s", err))
+	}
+}
+
+// ArtifactManifest defines an ocischema artifact manifest.
+type ArtifactManifest struct {
```

Comment:
Because this struct needs to satisfy the manifest interface, also there are some compatibility issues as v1.Artifact uses oci.Descriptor but distribution uses distribution.Descriptor.

Reply:
Changed subject to pointer after offline discussion. This gives `func (m Manifest) References()` a better design since it's much easier to decide whether it is nil.